### ISSUE TITLE
Msbuild will not honor configuration for nested project references when building solution file

### ISSUE THREAD

**Issue by shahabhijeet** [microsoft/msbuild#1556]
Repro:
1) Have a my.sln with ProjectA, ProjectB and have the project delayed signed for "Release" configuration
2) In ProjectB have a project reference for ProjectZ
3) All project default configuration is debug for this repro
3) Build my.sln with msbuild my.sln /p:Configuration=Release

Expected:
1) My.sln get's build without an error

Actual:
2) Project B fails to build, because ProjectZ is still getting build as "debug" and is not delay signed.

Investigation:
1) Seems msbuild creates the build map (meta project) with only the project listed in the solution with it's appropriate configuration
2) During this, any project not listed in solution will not be part of the meta project and hence will be defaulted to default settings during compilation.
3) AssignProjectConfiguration tasks caches solution projects, but nested project references (Project B referencing Project Z) are not part of the cache, hence when it comes to assigning project configuration, it does not find in the cache and hence defaults to project configuration (in this case Debug)

**Comment by rainersigwald** [comment]
Team triage: this is an unfortunate behavior. We wish the behavior for a `ProjectReference` to a not-in-the-solution project was to inherit the `Configuration` of the project that references it, to avoid this confusion.

However, this could result in confusing behavior of its own: the not-in-solution project could be built multiple times (in parallel) with different configurations.

Since we currently build only once, adding the possibility of parallel builds (and just changing the configuration of the not-in-solution project) could break users, so we're unwilling to make this change now.

The current behavior can be worked around by explicitly specifying the configuration using `SetConfiguration` metadata or by adding the referenced project to the solution.
